File: mini-core/src/main/java/com/mini/core/security/digest/SHA1.java
package com.mini.core.security.digest;

import java.nio.charset.Charset;

public final class SHA1 extends BaseDigest {
	public SHA1() {
		super("SHA-1");
	}
	
	public static SHA1 getInstance() {
		return new SHA1();
	}
	
	public static String encode(String data, Charset charset) {
		return getInstance().update(data, charset).encode();
	}
	
	public static String encode(String data) {
		return getInstance().update(data).encode();
	}
}

File: mini-util/src/main/kotlin/com/mini/core/security/digest/SHA1.kt
@file:Suppress("unused")

package com.mini.core.security.digest

import java.nio.charset.Charset

class SHA1 : BaseDigest("SHA-1") {
    companion object {
        @JvmStatic
        val instance: SHA1
            get() = SHA1()

        @JvmStatic
        fun encode(data: String?, charset: Charset?): String {
            return instance.update(data!!, charset!!).encode()
        }

        @JvmStatic
        fun encode(data: String?): String {
            return instance.update(data!!).encode()
        }
    }
}